core/common/src/main/java/alluxio/recorder/Recorder.java

```diff
@@ -24,62 +24,35 @@
 @NotThreadSafe
 public class Recorder {
   // To prevent adding too many records, causing memory leaks, set a maximum number of records
-  private static final int MAX_RECORDS_COUNT = 10000;
+  private static final int MAX_RECORDS_COUNT = 10_000;
 
   private final LinkedList<String> mRecords;
-  private boolean mEnableRecord;
-
-  private Recorder(LinkedList<String> records, boolean enable) {
-    mRecords = records;
-    mEnableRecord = enable;
-  }
 
   /**
-   * Create a disabled Recorder Object.
-   * By default, mEnableRecorder is false needs to be enabled by {@link Recorder#setEnabled()}.
-   * @return A {@code Recorder} Object
+   * Constructs a new instance.
    */
-  public static Recorder createDisabledRecorder() {
```

Comment:
keep this static factory method, and make `NoopRecorder` a singleton.

```java
public class NoopRecorder {
  /* package-private */ static final Recorder INSTANCE = new NoopRecorder();
}

public static Recorder createDisabledRecorder() {
  return NoopRecorder.INSTANCE;
}
```

Reply:
`NoopRecorder` can be made an private inner class of `Recorder` after this.

Reply:
Create a `NoopRecorder` instance by `Recorder.createNoopRecorder()`

Reply:
let's just call the method `public static Recorder createNoopRecorder()`

Reply:
yes, here is called like this
```java
return createWithRecorder(path, ufsConf, Recorder.createNoopRecorder());
```